### PR TITLE
JSON:API add inclusion of resources from path

### LINKED ISSUE #2365: Support inclusion of resources from a path according to JSON:API spec
Although [Inclusion of Related Resources](https://jsonapi.org/format/#fetching-includes) was partly implemented in #1949 as requested by #1934, unfortunately support for inclusion of resources from a path was not implemented. Would it be possible for this to be supported as well?

The following already works as expected (comments are included in the response):

```
GET /articles/1?include=comments HTTP/1.1
Accept: application/vnd.api+json
```

This doesn't work (neither comments nor authors are included in the response as they should be according to the spec):

```
GET /articles/1?include=comments.author HTTP/1.1
Accept: application/vnd.api+json
```

### PR DESCRIPTION
**Pull request by andreasschacht**
| Q             | A
| ------------- | ---
| Bug fix?      | no 
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tickets       | fixes #2365 
| License       | MIT
| Doc PR        | 

As specified in the JSON:API specs (https://jsonapi.org/format/#fetching-includes): In order to request resources related to other resources, a dot-separated path for each relationship name can be specified

> GET /articles/1?include=comments.author HTTP/1.1
Accept: application/vnd.api+json

### CODE REVIEW

File: src/JsonApi/Serializer/ItemNormalizer.php [hunk 1]
```diff
@@ -92,6 +92,19 @@ public function normalize($object, $format = null, array $context = [])
...
         $relationshipsData = $this->getPopulatedRelations($object, $format, $populatedRelationContext, $allRelationshipsData);
+
+        if (!isset($context['api_included_resources'])) {
+            // initialize tracking of already included resources
+            $context['api_included_resources'] = [];
+        }
+
+        // do not include primary resources
+        $context['api_included_resources'] = [
+            $resourceMetadata->getShortName() => [
```
Comment (dunglas) [inline src/JsonApi/Serializer/ItemNormalizer.php]: What about self-referencing relationships?

Comment (andreasschacht) [reply src/JsonApi/Serializer/ItemNormalizer.php]: Sorry, can't follow you. Maybe the comment above i misleading. It's just about primary data is already part of the "regular" answer and it's not necessary to put them into the `included` part too.

Comment (andreasschacht) [reply src/JsonApi/Serializer/ItemNormalizer.php]: @dunglas Do you want me to do or change anything?

**Comment by simonihmig** [comment]
Anything blocking this?

**Comment by reggermont** [comment]
Any news for this one ?

**Review by alanpoulain — APPROVED** [review]

**Comment by alanpoulain** [comment]
Nice improvement! I've simplified a little the code :slightly_smiling_face:

**Comment by alanpoulain** [comment]
Thank you very much @andreasschacht. Sorry for the time we took to merge your PR.

**Comment by rvmourik** [comment]
This is a nice improvement! Is it known in which version this will be included?

**Comment by alanpoulain** [comment]
In version `2.7`. We don't have a release date.

**Comment by rvmourik** [comment]
> In version `2.7`. We don't have a release date.

thanks for the quick reply

**Comment by Giant-Jelly** [comment]
image

**Pull request merged**